gulpfile.js

```diff
@@ -72,56 +80,74 @@ const browserifyConfigs = [
 const jsGlob = `./\{${dirs.join(',')}\}/**/*.js`;
 const jsGlobIgnore = `./\{${symlinkDirs.concat(copyDirs).join(',')}\}/**/*.js`;
 
-function onError(err) {
-	gutil.log(gutil.colors.red('Error:'), err);
-	this.emit('end');
+function onError(shouldExit, err) {
+	if(shouldExit) {
+		isExiting = true;
+		killAllWorkers();
+		throw new Error(err);
+	} else {
+		gutil.log(gutil.colors.red('Error:'), err);
+		this.emit('end');
+	}
 }
 
 function onSuccess(msg) {
-	gutil.log(gutil.colors.green('Build:'), msg);
+	if(!isExiting) {
+		gutil.log(gutil.colors.green('Build:'), msg);
+	}
 }
 
-function getBrowserify() {
+function getBrowserify(exitOnError = true) {
 	const streams = browserifyConfigs.map(config => {
 		return gulp
 			.src(config.src)
 			.pipe(tap(file => {
 				file.contents = browserify(file.path, config.config).bundle();
 			}))
 			.pipe(rename(config.dest))
+			.on('error', partial(onError, exitOnError))
```

Comment:
Oops, sorry, didn't look at the onError definition. Still, I'd prefer the replacement version — it seems clearer to me what it's doing without the extra dependency and without having to look up what `partial` does (in a library that's otherwise pretty much unneeded in the ES6+ era).